manifests/0000_70_dns-operator_02-deployment.yaml

```diff
@@ -36,6 +36,30 @@ spec:
         resources:
           requests:
             cpu: 10m
+      - name: kube-rbac-proxy
```

Comment:
This is super heavyweight vs just importing the library-go filters.  Are you not based on library-go?

Reply:
In general no one should be using rbac proxy for metrics if you are go code we own.

Reply:
We are based on controller-runtime. Not sure about what library-go offers here, but seems too late anyway. Is there a reason we shouldn't ship what we have?

Reply:
>In general no one should be using rbac proxy for metrics if you are go code we own.

cc @Miciah 

Clearly there's some communications breakdown on this subject. Do we need to revert this or not?

Reply:
To be clear, I don't know of any functional bugs with what we have

Reply:
Based on discussions with the operator-sdk team, I understood kube-rbac-proxy to be the correct approach.  I'm am unfamiliar with what library-go provides in this area.